test/src/main/java/org/apache/accumulo/test/functional/FateConcurrencyIT.java

```diff
@@ -322,25 +352,38 @@ public void getFateStatus() {
    */
   private boolean blockUntilCompactionRunning(final String tableName) {
 
-    int runningCompactions = 0;
+    long maxWait = defaultTimeoutSeconds() <= 0 ? 60_000 : ((defaultTimeoutSeconds() * 1000) / 2);
+
+    long startWait = System.currentTimeMillis();
 
     List<String> tservers = connector.instanceOperations().getTabletServers();
 
     /*
-     * wait for compaction to start - The compaction will acquire a fate transaction lock that used
-     * to block a subsequent online command while the fate transaction lock was held.
+     * wait for compaction to start on table - The compaction will acquire a fate transaction lock
+     * that used to block a subsequent online command while the fate transaction lock was held.
      */
-    while (runningCompactions == 0) {
+    while (System.currentTimeMillis() < (startWait + maxWait)) {
 
       try {
 
+        int runningCompactions = 0;
+
         for (String tserver : tservers) {
           runningCompactions += connector.instanceOperations().getActiveCompactions(tserver).size();
           log.trace("tserver {}, running compactions {}", tservers, runningCompactions);
         }
 
+        if (runningCompactions > 0) {
+          // Validate that there is a compaction fate transaction - otherwise test is invalid.
+          if (findFate(tableName)) {
+            return true;
+          }
+        }
+
       } catch (AccumuloSecurityException | AccumuloException ex) {
         throw new IllegalStateException("failed to get active compactions, test fails.", ex);
+      } catch (KeeperException ex) {
+        log.trace("Saw possible transient zookeeper error");
```

Comment:
Or maybe info... just saying that this is OK but I did see this exception.

Reply:
This is not an abnormal condition - it's just the way that zookeeper works and that it throws an exception to signal that a zk node no longer exists - sometimes, that could be an error if code was depending on something that should be there, other times, it's just that the node was removed by other threads or processes as a normal operation.  The logging at trace was so that if anyone was specifically interested logging could be enabled, but normally it's just zookeeper being zookeeper - the major change was to stop failing the test if this "normal", but transient  condition occurred, during the test.